packages/editor/src/components/header/index.js

```diff
@@ -82,21 +83,29 @@ function Header( {
 		};
 	}, [] );
 
-	const hasTopToolbar = isLargeViewport && hasFixedToolbar;
-
 	const [ isBlockToolsCollapsed, setIsBlockToolsCollapsed ] =
 		useState( true );
 
+	const hasCenter = isBlockToolsCollapsed && ! isTooNarrowForDocumentBar;
+	const hasBackButton = useHasBackButton();
+
 	// The edit-post-header classname is only kept for backward compatibilty
 	// as some plugins might be relying on its presence.
 	return (
-		<div className="editor-header edit-post-header">
-			<motion.div
-				variants={ backButtonVariations }
-				transition={ { type: 'tween' } }
-			>
-				<BackButton.Slot />
-			</motion.div>
+		<div
+			className={ clsx( 'editor-header edit-post-header', {
+				'has-back-button': hasBackButton,
+				'has-center': hasCenter,
```

Comment:
Do we really need this kind of classes? Can't we figure out a unified layout that works without requiring these?

Reply:
Potentially more advanced and probably less readable selectors in the CSS could be used. I have another design change I’d like to propose for the header that should eliminate the need for `has-center` and simplify some CSS but it’s too distinct for this PR.

I’ll take another look anyway. I don’t love the classes either.

Reply:
Okay, done in 574b8bc4c529e17c3c70d10cd1543f671756c37c and 884c534862dfb78a21f20f235bf49c9a38a79d75. I still don’t love using `:has()` but perhaps I’ll be able to get rid of it with the aforementioned change I’ll propose in a later PR.